src/main/java/uk/gov/hmcts/darts/audio/service/impl/MediaRequestServiceImpl.java

```diff
@@ -500,19 +503,31 @@ public MediaPatchResponse patchMediaRequest(Integer mediaRequestId, MediaPatchRe
     }
 
     @Override
-    public List<AdminMediaSearchResponseItem> adminMediaSearch(Integer transformedMediaId) {
-        Optional<TransformedMediaEntity> transformedMediaOpt = transformedMediaRepository.findById(transformedMediaId);
-        if (transformedMediaOpt.isEmpty()) {
+    public List<AdminMediaSearchResponseItem> adminMediaSearch(Integer transformedMediaId, List<Integer> hearingIds, OffsetDateTime startAt,
+                                                               OffsetDateTime endAt) {
+        MediaSearchData searchData = new MediaSearchData(transformedMediaId, hearingIds, startAt, endAt);
+        searchMediaValidator.validate(searchData);
+
+        List<TransformedMediaEntity> mediaRequestEntities = transformedMediaRepository
+            .findTransformMediaWithStartAndEndDateTimeRange(transformedMediaId, hearingIds, startAt, endAt);
```

Comment:
Not sure this is correct - we want to locate `media` objects by their `start_ts` and `end_ts` attributes, whereas this is locating `media_request` objects. I think the `adminMediaSearch()` might need some adjusting to cater for that.

Reply:
The original logic finds the media request by the transformed id and then filers the media based on the date range. This seems to be due to there being no direct association between media request and media. I will have a think.

Reply:
Maybe we need two code flows - one that caters to `transformed_media_id` being sent in the request (using the existing query that originates at the `media_request`), and a separate flow/query catering to the new parameters that originates at `media`?